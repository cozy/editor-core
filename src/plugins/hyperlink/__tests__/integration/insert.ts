import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import {
  goToEditorTestingWDExample,
  mountEditor,
} from '../../../../__tests__/__helpers/testing-example-helpers';
import {
  fullpage,
  editable,
  getDocFromElement,
  quickInsert,
} from '../../../../__tests__/integration/_helpers';
import {
  clickToolbarMenu,
  ToolbarMenuItem,
} from '../../../../__tests__/__helpers/page-objects/_toolbar';
import { hyperlinkSelectors } from '../../../../__tests__/__helpers/page-objects/_hyperlink';

BrowserTestCase(
  'can insert hyperlink with only URL using toolbar',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
    });

    await clickToolbarMenu(page, ToolbarMenuItem.link);
    await page.waitForSelector(hyperlinkSelectors.linkInput);
    await page.type(hyperlinkSelectors.linkInput, 'http://atlassian.com');
    await page.keys('Return');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'can insert hyperlink with URL and text using toolbar',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
    });

    await clickToolbarMenu(page, ToolbarMenuItem.link);
    await page.waitForSelector(hyperlinkSelectors.linkInput);
    await page.type(hyperlinkSelectors.linkInput, 'http://atlassian.com');
    await page.type(hyperlinkSelectors.textInput, 'Atlassian');
    await page.keys('Return');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'can insert hyperlink via quick insert menu',
  {},
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
    });

    await quickInsert(page, 'Link');
    await page.waitForSelector(hyperlinkSelectors.linkInput);
    await page.type(hyperlinkSelectors.linkInput, 'http://atlassian.com');
    await page.type(hyperlinkSelectors.textInput, 'Atlassian');
    await page.keys('Return');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
