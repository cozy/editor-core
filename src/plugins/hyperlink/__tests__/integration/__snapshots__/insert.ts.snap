// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can insert hyperlink via quick insert menu 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Atlassian",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`can insert hyperlink with URL and text using toolbar 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Atlassian",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`can insert hyperlink with only URL using toolbar 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "http://atlassian.com",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
